node/resolvers/search/index.ts

```diff
@@ -432,13 +432,31 @@ export const queries = {
       hideUnavailableItems: args.hideUnavailableItems,
     }
 
-    const result = await biggySearch.facets(biggyArgs)
+    const facetPromises = [biggySearch.facets(biggyArgs)]
+
+    const showCategoryTree = args.categoryTreeBehavior === 'show';
+    const categorySelectedFacets = args.selectedFacets.filter(facet => facet.key === 'c')
```

Comment:
What happens if the selected facet's are `category-*`, should we handle those cases?

Reply:
Well noticed! It would not filter those facets. Fixed!